desktop/core/src/desktop/js/apps/storageBrowser/StorageBrowserPage/StorageBrowserPage.tsx

```diff
@@ -14,19 +14,54 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Tabs, Spin } from 'antd';
+import type { TabsProps } from 'antd';
 
-import { i18nReact } from '../../../utils/i18nReact';
+import DataBrowserIcon from '@cloudera/cuix-core/icons/react/DataBrowserIcon';
 
+import { i18nReact } from '../../../utils/i18nReact';
 import CommonHeader from '../../../reactComponents/CommonHeader/CommonHeader';
-import DataBrowserIcon from '@cloudera/cuix-core/icons/react/DataBrowserIcon';
+import StorageBrowserTabContent from './StorageBrowserTabContents/StorageBrowserTabContent';
+import { fetchFileSystems } from '../../../reactComponents/FileChooser/api';
+
+import './StorageBrowserPage.scss';
 
 const StorageBrowserPage: React.FC = (): JSX.Element => {
+  const [fileSystemTabs, setFileSystemTabs] = useState<TabsProps['items'] | undefined>();
+  const [loading, setLoading] = useState(true);
+
   const { t } = i18nReact.useTranslation();
 
+  useEffect(() => {
+    if (!fileSystemTabs) {
+      setLoading(true);
+      fetchFileSystems()
+        .then(fileSystems => {
+          const fileSystemsObj = fileSystems.map((system, index) => {
+            return {
+              label: system.file_system.toUpperCase(),
+              key: index.toString(),
```

Comment:
key should be unique for tabs, here the keys will be 1,2,3 so i might get redundant. use `system.file_system+'_tab'` or something different.